testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/rbac/JmxRBACProviderServerGroupScopedRolesTestCase.java

```diff
@@ -0,0 +1,311 @@
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.domain.rbac;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.PROFILE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.jboss.as.test.integration.domain.rbac.AbstractServerGroupScopedRolesTestCase.USERS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.management.Attribute;
+import javax.management.JMRuntimeException;
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+import javax.management.remote.JMXServiceURL;
+import org.apache.commons.lang.ArrayUtils;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.client.helpers.domain.DomainClient;
+import org.jboss.as.network.NetworkUtils;
+import org.jboss.as.test.integration.domain.suites.FullRbacProviderTestSuite;
+import org.jboss.as.test.integration.management.interfaces.JmxManagementInterface;
+import org.jboss.as.test.integration.management.rbac.RbacAdminCallbackHandler;
+import org.jboss.as.test.integration.management.rbac.UserRolesMappingServerSetupTask;
+import org.jboss.dmr.ModelNode;
+import org.jboss.sasl.util.UsernamePasswordHashUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.test.jmx.JMXServiceDeploymentSetupTask;
+
+/**
+ * Tests of server group scoped roles using the "rbac" access control provider.
+ *
+ * @author Brian Stansberry (c) 2013 Red Hat Inc.
+ */
+public class JmxRBACProviderServerGroupScopedRolesTestCase extends AbstractServerGroupScopedRolesTestCase {
```

Comment:
This test validates the expected behavior of server-group-scoped roles when the user is in a role that allows access to the server (i.e. it's in the group). But it's not validating that users in scoped roles whose server groups don't include the target server are rejected.

We should add a test for host-scoped roles as well, as the same basic pattern should be covered.